osu.Game/Screens/Play/IGameplayClock.cs

```diff
@@ -23,6 +23,14 @@ public interface IGameplayClock : IFrameBasedClock
         /// </summary>
         IAdjustableAudioComponent AdjustmentsFromMods { get; }
 
+        /// <summary>
+        /// Whether gameplay is paused.
+        /// </summary>
         IBindable<bool> IsPaused { get; }
+
+        /// <summary>
+        /// Whether the clock is currently rewinding.
+        /// </summary>
+        bool IsRewinding { get; }
```

Comment:
So I get why this is separated, but my main question is what should happen to places that reinvent it, and there are a fair few and I haven't even searched that thoroughly. Some do it closer to how the implementation added here do:

https://github.com/ppy/osu/blob/001d09057ed76e29ccedefa95ff7a5c96e2e3c83/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderBall.cs#L190-L191

some less so:

https://github.com/ppy/osu/blob/001d09057ed76e29ccedefa95ff7a5c96e2e3c83/osu.Game/Screens/Play/ComboEffects.cs#L56-L57
https://github.com/ppy/osu/blob/001d09057ed76e29ccedefa95ff7a5c96e2e3c83/osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchComboCounter.cs#L62-L68
https://github.com/ppy/osu/blob/001d09057ed76e29ccedefa95ff7a5c96e2e3c83/osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs#L300-L302
https://github.com/ppy/osu/blob/001d09057ed76e29ccedefa95ff7a5c96e2e3c83/osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs#L690-L692

It seems like it would be a good idea to migrate everything across to use `IsRewinding`, to avoid local (and potentially wrong) reimplementation, but that _also_ means that some of the above implementations that worked with non-gameplay clocks (or without one at all, depending on how required we make `GameplayClock` DI) may cease to do so.

Reply:
I did mean to do this but it slipped my mind.